python/whylogs/api/logger/segment_processing.py

```diff
@@ -44,6 +46,13 @@ def _get_segment_from_group_key(group_key, partition_id) -> Tuple[str, ...]:
     return Segment(segment_tuple_key, partition_id)
 
 
+def _is_nan(x):
```

Comment:
if the data being processed is a pandas series, there is a hasnans property that we can use. The isnan check is surprisingly slow when we tested this in the past, so if there is a way to avoid checking each item it will save a lot on compute costs of profiling large inputs.

Reply:
this is not being done on a pandas series. It's only on all the possible combinations of segments, not on the raw data. We can try to look for more efficient ways, but I think it's not such a bottleneck, considering the number of checks is limited by the number of segments? or rather, number of groups times the number of elements in a group.

Reply:
Thank you for that clarification, makes a lot more sense now that I understand this is only used on the smaller set of segments.